batcher/aligned/src/main.rs

```diff
@@ -107,20 +109,87 @@ pub struct SubmitArgs {
     keystore_path: Option<PathBuf>,
     #[arg(name = "Private key", long = "private_key")]
     private_key: Option<String>,
-    #[arg(
-        name = "Max Fee (ether)",
-        long = "max_fee",
-        default_value = "0.0013ether" // 13_000 gas per proof * 100 gwei gas price (upper bound)
-    )]
-    max_fee: String, // String because U256 expects hex
+    #[command(flatten)]
+    price_estimate: PriceEstimateArgs,
     #[arg(name = "Nonce", long = "nonce")]
     nonce: Option<String>, // String because U256 expects hex
     #[arg(
         name = "The working network's name",
         long = "network",
         default_value = "devnet"
     )]
-    network: NetworkArg,
+    network: Network,
+}
+
+#[derive(Args, Debug)]
+#[group(required = false, multiple = false)]
+pub struct PriceEstimateArgs {
+    #[arg(
+        name = "Max Fee (ether)",
+        long = "max_fee",
+        help = "Specifies the maximum fee (`max_fee`) the user is willing to pay for the submitted proof, in Ether."
+    )]
+    max_fee: Option<String>, // String because U256 expects hex
+    #[arg(
+        name = "NUMBER_PROOFS_IN_BATCH",
+        long = "custom_fee_estimate",
+        help = "Specifies a `max_fee` equivalent to the cost of paying 1 proof / `num_proofs_in_batch` allowing the user a user to estimate there `max_fee` precisely based on the `number_proofs_in_batch`."
+    )]
+    custom_fee_estimate: Option<usize>,
+    #[arg(
+        name = "Price Estimate: Instant",
```

Comment:
here for example, 'Price Estimate: Instant' doesn't make much sense of out context.